myaws/autoscaling_set_instance_protection.go

```diff
@@ -0,0 +1,48 @@
+package myaws
+
+import (
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/pkg/errors"
+	"math"
+)
+
+// AutoScalingSetInstanceProtectionOptions customizes the behavior of the Attach command.
+type AutoScalingSetInstanceProtectionOptions struct {
+	AsgName              string
+	InstanceIds          []*string
+	ProtectedFromScaleIn bool
+}
+
+// AutoScalingSetInstanceProtection protects from termination when scale in your autoscaling group.
+func (client *Client) AutoScalingSetInstanceProtection(options AutoScalingSetInstanceProtectionOptions) error {
+	// the number of maximum InstanceIds is limited to 19.
+	// https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_SetInstanceProtection.html
+	maxInstanceIDCount := 19
+	var maxIteration = int(math.Ceil(float64(len(options.InstanceIds)) / float64(maxInstanceIDCount)))
+	for i := 0; i < maxIteration; i++ { // set AutoScalingInstanceProtection every 19 instances.
```

Comment:
The myaws has been already depended on `go-funk`. So we can easily write this kind of paging loop using `funk.Chunk`. An example can be found in `ECSNodeDrain`

https://github.com/minamijoyo/myaws/blob/614f0e90bdc658dcbe787dfbe6d462ab73a20bbf/myaws/ecs_node_drain.go#L28-L40

Reply:
Thank you telling me nice library!
I refactored this.